### ISSUE TITLE
Pyqtgraph is neglected

### ISSUE THREAD

**Issue by sbourdeauducq** [pyqtgraph/pyqtgraph#238]
Last commit on Aug 9 and even simple pull requests are not merged.

**Comment by campagnola** [comment]
:(  Sorry, everyone! 
I've been very busy moving my family across the country to start my new job. Things should ease up in a few weeks. Hang in there, and feel free to contact me directly if you'd like to discuss how we can get more people keys to the github repository.

**Comment by jdreaver** [comment]
Thanks for your work @campagnola! I'm a happy day-to-day pyqtgraph user, and I really appreciate your efforts both here and on vispy. Good luck with the new job, and please don't let your open source work cause you too much stress! :)

**Comment by sbourdeauducq** [comment]
Are there any volunteers to pick up maintenance? @jdreaver ?

**Comment by kernc** [comment]
It would probably make sense for @campagnola to make another release before any such radical maintenance (:+1:) is introduced.

**Comment by jkfindeisen** [comment]
@sbourdeauducq There surely will be volunteers. I would be one for example. But there is more to do than just a fork of pyqtgraph and keeping it uptodate. There is the website http://pyqtgraph.org/ and there is the pypi site: https://pypi.python.org/pypi/pyqtgraph. So without @campagnola either getting more active or sharing credentials it's difficult to imagine any easy improvement.

**Comment by campagnola** [comment]
I am happy to start giving out the keys. The thing I am most uncertain about is to whom the keys should be given--ideally these would be people who have demonstrated involvement with the community in the past. For github push privileges, I am looking for people who have demonstrated an understanding of the architecture and an ability to think critically about shaping the API. I have already identified several people I'd like to get involved in code maintenance, but I haven't had much time to engage with them yet.

@kernc, whoever gets involved will need to know how to make releases. It would make sense for them to be involved in the next release.

**Comment by jdreaver** [comment]
I want to jump ship to vispy soon for my Python plotting needs, and I'm also working extensively on a scenegraph/plotting framework in OpenGL for Haskell. However, I still use pyqtgraph extensively at work.

@campagnola, are you planning on using a vispy as a backend for pyqtgraph, or deprecating pyqtgraph in favor of vispy?

I also agree that a new release would be good before any sweeping changes.

Also, as an FYI on the how awesome pyqtgraph is: I made a presentation yesterday to a lot of oil industry engineers. I work on scientific simulation software, and most of our competitors are using very old Java GUIs. I showed them our plots made with pyqtgraph, and they were blown away with how good they look and how fast they are with huge data sets. Thanks for your work!

**Comment by campagnola** [comment]
There are many features in PyQtGraph that are out of scope for VisPy, so complete deprecation is not an option. I still use PyQtGraph daily in my work and will continue to do so for the forseeable future. 

In the short term I plan to deprecate pyqtgraph.opengl. In the long term, I would like to see VisPy replace Qt's GraphicsView as the backend on which pyqtgraph draws its plots, but that remains a distant dream unless someone shows up with the resources and the need for such a thing.

**Comment by douzepouze** [comment]
Thank you for your awesome work!

**Comment by stxlvt** [comment]
@campagnola What is the current state of 3d GL plotting with python? I noticed pyqtgraph.opengl hasn't been touched for 2 years, and you mention wanting to deprecate it in this issue. Also, from the looks of the [vispy contribution graph](https://github.com/vispy/vispy/graphs/contributors), not much is happening there either.

What would you recommend using when starting a new project with 3d visualization needs in python?

**Comment by efayadspecific** [comment]
Hi - and thanks for creating this awesome library!
That's the same question I have as well, but not necessarily from a 3d perspective, but from the point of view of general high performance interactive Python visualization.  I understand the difference in scope between VisPy and PyQtGraph, but in terms of starting out with the use of either, is using one more risky, in terms of its maintainability?

**Comment by efayadspecific** [comment]
And Glumpy seems to be listed as a sister project of VisPy, so adding that to the mix, any recommendations?

**Comment by campagnola** [comment]
Current status of things, as I see them:
* PyQtGraph continues to be more mature than VisPy; for now I think this is the safer bet unless you are certain that you will need GPU support.
* For 3D / GPU-accelerated work, I personally use VisPy rather than pyqtgraph.opengl because I love its architecture--the foundations it has with gloo, shader generation, and transforms beat any other 3D system I have worked with.
* However, VisPy still has many rough edges and is sorely lacking in support, so there is a bit of risk there. It's easier for me since I can patch up the holes in VisPy as I come across them.
* Last I checked, @rougier is still developing Glumpy, although he considers it his "playground"; I think he would be the one to ask about his future plans there. I have not spent much time with it.

In short, there are still no clear winners in the python + opengl world. I am still hoping VisPy can come through, but I think that will only happen with the dedicated attention of a paid developer.

**Comment by efayadspecific** [comment]
Thank you for that.  Much appreciated.

**Comment by rougier** [comment]
I still maintain glumpy and I'm in the process of writing an (open access) book about Python/OpenGL/Scientific visualization. glumpy is more low level than vispy or pyqtgraph though.

**Comment by Dschoni** [comment]
So, this is two years later and I haven't seen much going on. Can we at least get the Travis-CI going again, so that a (possible future) maintainer can check test coverage of pull-requests? I am willing to put some of my time into getting pyqtgraph back on track with new numpy/qt/python versions, but I am far from being a good maintainer ;)

**Comment by campagnola** [comment]
If you're interested in helping, there is a small gathering of volunteers going on in pyqtgraph.slack.com. Sene me a message at luke.campagnola@gmail.com and I'll send you an invite.

**Comment by Dschoni** [comment]
Hei @campagnola I tried to write you an e-mail but it doesn't seem to reach you. Can you try reaching me on github@dschoni.de ?

**Comment by j9ac9k** [comment]
I'm going to close out this issue as part of the clean up effort, we've gotten some more contributors, we got travis cleaned up (still have some test failures on py27) and we have pretty extensive azure pipelines setup going.  I can't speak for Luke but we would certainly welcome more contributors, there is a huge backlog of issues/PRs to get through.
